### PR TITLE
Add a docs linter

### LINKED ISSUE #2407: Add a linter for `docs/`
Previously discussed here: https://github.com/unitaryfund/mitiq/pull/2382#discussion_r1633795704

Option 1: There exists [a Sphinx extension for `.rst` files](https://github.com/sphinx-contrib/sphinx-lint). Will require additional configuration as most of our examples and user guides are written in `.md` format

Option 2: [`markdownlint`](https://github.com/DavidAnson/markdownlint) for `.md` files has a nice set of pre-defined [rules](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md). We can also define our own [rules](https://github.com/markdownlint/markdownlint/blob/41fc308f0d7f2647f0ae2c2d3826f48e42d964f4/docs/creating_rules.md) as needed.

### PR DESCRIPTION
**Pull request by purva-thakre**
## Description

Fixes #2407 

To check this action locally, install [`nvm`](https://stackoverflow.com/a/39981888) in your local virtual environment. 

Then, install `markdownlint-cli` using the [provided instructions](https://github.com/igorshubovych/markdownlint-cli?tab=readme-ov-file#installation). Run `markdownlint docs/` to find all the rule violations.

---

### License

- [ ] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/main/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -66,6 +66,9 @@ jobs:
         with:
           path: ./docs/build/.jupyter_cache
           key: jupyter-cache-${{ steps.gen_hash.outputs.hash }}
+      
+      - name: Lint docs
+        uses: DavidAnson/markdownlint-cli2-action@v16
```
Comment (purva-thakre) [inline .github/workflows/build.yml:71]: right now, this is only finding 5 `.md` files.

File: docs/source/examples/rshadows_tutorial.md [hunk 2]
```diff
@@ -147,43 +144,45 @@ def cirq_executor(
...
+
 The classical shadow estimation involves Pauli twirling of a quantum channel represented by $\mathcal{G} \subset U(d)$, with PTM representation $\mathcal{U}$. This twirling allows direct computation of $\widehat{\mathcal{M}}$ for the noisy channel $\Lambda$:
 \begin{equation}
-\widehat{\mathcal{M}} = \mathbb{E}_{\mathcal{G}}[\mathcal{U}^\dagger\mathcal{M}_z\Lambda\mathcal{U}]
+\widehat{\mathcal{M}} = \mathbb{E}*{\mathcal{G}}[\mathcal{U}^\dagger\mathcal{M}_z\Lambda\mathcal{U}]
 \end{equation}
 Local Clifford group projections are given by:
 \begin{equation}
-\Pi_{b_i}=\left\{
+\Pi*{b_i}=\left\{
```
Comment (purva-thakre) [inline docs/source/examples/rshadows_tutorial.md:166]: If we decide to keep this tool, we want to make sure it does not make changes to LaTex equations.

**Comment by natestemen** [comment]
I don't love the idea of adding node as a dependency to Mitiq. A simpler tool that can get us part of the way there is https://editorconfig.org/, and it's supported by almost all editors. Setting `trim_trailing_whitespace` to true would fix some of these issues. The drawback here is that this tool is not markdown specific.

**Comment by purva-thakre** [comment]
Closing this PR to cherry-pick the changes flagged by this extension with a different PR.

**Pull request closed without merging**